pylint/lint.py

```diff
@@ -89,6 +90,24 @@
 MANAGER = astroid.MANAGER
 
 
+def _ast_from_string(data, filepath, modname):
+    if (
+        modname in MANAGER.astroid_cache
+        and MANAGER.astroid_cache[modname].file == filepath
+    ):
+        return MANAGER.astroid_cache[modname]
+    from astroid.builder import AstroidBuilder
+
+    return AstroidBuilder(MANAGER).string_build(data, modname, filepath)
+
+
+def _read_stdin():
+    # https://mail.python.org/pipermail/python-list/2012-November/634424.html
+    # FIXME should this try to check the file's declared encoding?
+    sys.stdin = TextIOWrapper(sys.stdin.detach(), newline=None)
```

Comment:
I don't yet understand why we need this line.

Reply:
This probably helps: https://stackoverflow.com/questions/31488419/binary-input-with-an-ascii-text-header-read-from-stdin